### PR TITLE
Check in Cloud Job Discovery quick start

### PR DESCRIPTION
**Pull request by xinyunh0929**
CJD got Public beta launch so we would like to have our samples checked in as other projects.

### CODE REVIEW

**Comment by xinyunh0929** [comment]
I saw the other team writing the test using 'ava'. Could we use 'mocha' instead?

**Comment by ace-n** [comment]
IIRC we use Ava since it supports parallel test running - @jmdobry can confirm.

File: jobs/cjd_sample/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<img src="https://avatars2.githubusercontent.com/u/2810941?v=3&s=96" alt="Google Cloud Platform logo" title="Google Cloud Platform" align="right" height="96" width="96"/>
+
+# Google Cloud Job Discovery API Samples
+
+Cloud Job Discovery is part of Google for Jobs - a Google-wide commitment to help
```
Comment (ace-n) [inline jobs/cjd_sample/README.md]: Was this autogenerated using `repo-tools generate`? If not, it should be.

File: jobs/cjd_sample/quickstart.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// [START quickstart]
+
+// Imports the Google APIs client library
+const {google} = require('googleapis');
```
Comment (ace-n) [inline jobs/cjd_sample/quickstart.js]: Do we have a client library specifically for this, as opposed to `googleapis`? (Paging @theacodes)

File: jobs/cjd_sample/quickstart.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  });
+
+  // Lists companies
+  jobs.companies.list((err, result) => {
+    if (err) {
+      console.error(err);
+      return;
+    }
+
+    console.log('Request ID: ' + result.data.metadata.requestId);
```
Comment (ace-n) [inline jobs/cjd_sample/quickstart.js]: Two nits:
1) Use promises
2) Use template strings

File: jobs/cjd_sample/README.md (hunk 1 above)
Comment (xinyunh0929) [reply jobs/cjd_sample/README.md]: Do you mean the README should be generated from repo-tools generate?
Current README is written by myself. Should I remove it then run repo-tools generate?

File: jobs/cjd_sample/quickstart.js (hunk 3 above)
Comment (xinyunh0929) [reply jobs/cjd_sample/quickstart.js]: Sorry, I am not too familiar with nodeJs. How could I use promise in this case?

File: jobs/cjd_sample/README.md (hunk 1 above)
Comment (xinyunh0929) [reply jobs/cjd_sample/README.md]: generate: Generating lib_readme in: /usr/local/google/home/xinyunh/githubCode/nodeJsPersonalFork/nodejs-docs-samples/jobs/cjd_sample
generate: Compiling: /usr/local/google/home/xinyunh/githubCode/nodeJsPersonalFork/nodejs-docs-samples/jobs/cjd_sample/README.md
generate: Failed to generate: /usr/local/google/home/xinyunh/githubCode/nodeJsPersonalFork/nodejs-docs-samples/jobs/cjd_sample/README.md
generate: Oh no! Generating failed after 0.0230s.
generate: TypeError: Cannot read property 'toUpperCase' of undefined
    at Object.exports.createReleaseQualityBadge (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/cjs/utils/index.js:222:35)
    at Object.eval [as main] (eval at createFunctionContext (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:254:23), <anonymous>:12:88)
    at main (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/handlebars/dist/cjs/handlebars/runtime.js:175:32)
    at ret (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/handlebars/dist/cjs/handlebars/runtime.js:178:12)
    at ret (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:526:21)
    at /usr/lib/node_modules/@google-cloud/nodejs-repo-tools/cjs/cli/commands/generate.js:276:46
    at Array.forEach (<anonymous>)
    at Object.exports.handler (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/cjs/cli/commands/generate.js:238:16)
    at Object.runCommand (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/yargs/lib/command.js:235:44)
    at Object.parseArgs [as _parseArgs] (/usr/lib/node_modules/@google-cloud/nodejs-repo-tools/node_modules/yargs/yargs.js:1014:30)

File: jobs/cjd_sample/system-test/quickstart.test.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2017, Google, Inc.
```
Comment (happyhuman) [inline jobs/cjd_sample/system-test/quickstart.test.js]: 2018 LLC

File: jobs/cjd_sample/quickstart.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+// Acquires credentials
+google.auth.getApplicationDefault((err, authClient) => {
+  if (err) {
+    console.error('Failed to acquire credentials');
+    console.error(err);
+    return;
+  }
+
+  if (authClient.createScopedRequired && authClient.createScopedRequired()) {
```
[the remark this answers is not in the document]
Comment (xinyunh0929) [reply jobs/cjd_sample/quickstart.js]: Not sure. I just the follow the same way 'kms' did in their samples for authentication.

**Pull request merged**